extensions/kafka-eight-simpleConsumer/src/main/java/io/druid/firehose/kafka/KafkaEightSimpleConsumerFirehoseFactory.java

```diff
@@ -0,0 +1,318 @@
+{
+  private static final Logger log = new Logger(
+      KafkaEightSimpleConsumerFirehoseFactory.class
+  );
+
+  @JsonProperty
+  private final List<String> brokerList;
+
+  @JsonProperty
+  private final List<Integer> partitionIdList;
+
+  @JsonProperty
+  private final String clientId;
+
+  @JsonProperty
+  private final String feed;
+
+  @JsonProperty
+  private final int queueBufferLength;
+
+  @JsonProperty
+  private boolean earliest;
+
+  private final List<PartitionConsumerWorker> consumerWorkers = new CopyOnWriteArrayList<>();
+  private static final int defaultQueueBufferLength = 20000;
+  private static final String resetToLatest = "latest";
+  @JsonCreator
+  public KafkaEightSimpleConsumerFirehoseFactory(
+      @JsonProperty("brokerList") List<String> brokerList,
+      @JsonProperty("partitionIdList") List<Integer> partitionIdList,
+      @JsonProperty("clientId") String clientId,
+      @JsonProperty("feed") String feed,
+      @JsonProperty("queueBufferLength") Integer queueBufferLength,
+      @JsonProperty("resetBehavior") String resetBehavior
+  )
+  {
+    this.brokerList = brokerList;
+    this.partitionIdList = partitionIdList;
+    this.clientId = clientId;
+    this.feed = feed;
+
+    this.queueBufferLength = queueBufferLength == null ? defaultQueueBufferLength : queueBufferLength;
+    log.info("queueBufferLength loaded as[%s]", this.queueBufferLength);
+
+    this.earliest = resetToLatest.equalsIgnoreCase(resetBehavior) ? false : true;
+    log.info("Default behavior of cosumer set to earliest? [%s]", this.earliest);
+  }
+
+  private Map<Integer, Long> loadOffsetFromPreviousMetaData(Object lastCommit)
+  {
+    Map<Integer, Long> offsetMap = Maps.newHashMap();
+    if (lastCommit == null) {
+      return offsetMap;
+    }
+    if (lastCommit instanceof Map) {
+      Map<Object, Object> lastCommitMap = (Map) lastCommit;
+      for (Map.Entry<Object, Object> entry : lastCommitMap.entrySet()) {
+        try {
+          int partitionId = Integer.parseInt(entry.getKey().toString());
+          long offset = Long.parseLong(entry.getValue().toString());
+          log.debug("Recover last commit information partitionId [%s], offset [%s]", partitionId, offset);
+          offsetMap.put(partitionId, offset);
+        }
+        catch (NumberFormatException e) {
+          log.error(e, "Fail to load offset from previous meta data [%s]", entry);
+        }
+      }
+      log.info("Loaded offset map[%s]", offsetMap);
+    } else {
+      log.error("Unable to cast lastCommit to Map");
+    }
+    return offsetMap;
+  }
+
+  @Override
+  public FirehoseV2 connect(final ByteBufferInputRowParser firehoseParser, Object lastCommit) throws IOException
+  {
+    final Map<Integer, Long> lastOffsets = loadOffsetFromPreviousMetaData(lastCommit);
+
+    Set<String> newDimExclus = Sets.union(
+        firehoseParser.getParseSpec().getDimensionsSpec().getDimensionExclusions(),
+        Sets.newHashSet("feed")
```

Comment:
Yes, because I saw https://github.com/druid-io/druid/blob/master/extensions/kafka-eight/src/main/java/io/druid/firehose/kafka/KafkaEightFirehoseFactory.java#L71 the current KafkaEightFirehose doesn't allow `feed` as dimension, so I followed the same logic here